src/Instrumentation/Symfony/src/HttpClientInstrumentation.php

```diff
@@ -33,6 +33,11 @@ public static function register(): void
                 ?string $filename,
                 ?int $lineno,
             ) use ($instrumentation): array {
+                /** @psalm-suppress UndefinedClass */
+                if (class_exists("\ApiPlatform\Symfony\Bundle\Test\Client") && $client instanceof \ApiPlatform\Symfony\Bundle\Test\Client) {
```

Comment:
```suggestion
                if ($client instanceof \ApiPlatform\Symfony\Bundle\Test\Client) {
```

`class_exists()` triggers an unnecessary autoload attempt.

Condition has to be checked in `post` hook too, otherwise we might detach the parent scope / end the parent span.